### PR TITLE
Increase stdlib to v9. Increase concat to v9. Update PDK to v3

### PR DESCRIPTION
**Pull request by canihavethisone**
Unit tests pass with Puppet 7.25.  There is a fail with Puppet 8.2.0 so left metadata at v7.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -15,27 +15,35 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - ruby: 2.7.6
+          - ruby: 2.7.8
```
Comment (ghoneycutt) [inline .github/workflows/ci.yaml:18]: Why the version change here?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -15,27 +15,35 @@ jobs:
...
+            puppet: 7
+            fixtures: .fixtures-latest.yml
+            allow_failure: true
+          - ruby: 3.2.2
+            puppet: 7
+            fixtures: .fixtures.yml
+            allow_failure: false
+          - ruby: 3.2.2
```
Comment (ghoneycutt) [inline .github/workflows/ci.yaml:30]: Is puppet 7 running ruby 3.2.2?

File: spec/acceptance/nodesets/el8.yml [hunk 3]
```diff
@@ -1,22 +1,25 @@
 HOSTS:
-  rocky-8:
+  el8:
```
Comment (ghoneycutt) [inline spec/acceptance/nodesets/el8.yml:2]: Why the move from rocky to el?

**Comment by ghoneycutt** [comment]
This would be easier to understand if you separated stdlib 9, adding platforms and removing platforms into different PR's.

**Comment by canihavethisone** [comment]
Hey, they were changes made by pdk.  You can either correct them as required or i can replace the pr with one that doesn't upgrade pdk

**Pull request closed without merging**